components/providers/categories-modal-tanstack.tsx

```
@@ -0,0 +1,142 @@
+import { featuredCategories } from "./category-icons";
+
+// Define category groups
+const CATEGORY_GROUPS = {
+  AUTOMOTIVE: "Automotive",
+  CONSTRUCTION: "Construction & Building",
+  INDUSTRIAL: "Industrial & Equipment",
+  SERVICES: "Services & Maintenance"
+};
+
+// Function to assign categories to groups
+function categorizeByGroup(categories: any[]): Record<string, any[]> {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)